pkg/controller/migmigration/task.go

```diff
@@ -87,54 +104,123 @@ func (t *Task) Run() error {
 		return err
 	}
 
-	// Annotate persistent storage resources with actions
-	err = t.annotateStorageResources()
+	// Return unless restic restart has finished
+	if t.Phase == Started || t.Phase == WaitOnResticRestart {
+		return nil
+	}
+
+	// Run initial Backup if this is not a stage
+	// The initial backup captures the state of the applications on the source
+	// cluster while explicity setting `includePersistentVolumes` value to
+	// `false`. This will capture everything except PVs
+	if !t.stage() {
+		err = t.ensureInitialBackup()
+		if err != nil {
+			log.Trace(err)
+			return err
+		}
+		switch t.InitialBackup.Status.Phase {
+		case velero.BackupPhaseCompleted:
+			t.Phase = InitialBackupCompleted
+		case velero.BackupPhaseFailed:
+			reason := fmt.Sprintf(
+				"Backup: %s/%s failed.",
+				t.InitialBackup.Namespace,
+				t.InitialBackup.Name)
+			t.addErrors([]string{reason})
+			t.Phase = InitialBackupFailed
+			return nil
+		case velero.BackupPhasePartiallyFailed:
+			reason := fmt.Sprintf(
+				"Backup: %s/%s partially failed.",
+				t.InitialBackup.Namespace,
+				t.InitialBackup.Name)
+			t.addErrors([]string{reason})
+			t.Phase = InitialBackupFailed
+			return nil
+		case velero.BackupPhaseFailedValidation:
+			t.addErrors(t.InitialBackup.Status.ValidationErrors)
+			t.Phase = InitialBackupFailed
+			return nil
+		default:
+			t.Phase = InitialBackupStarted
+			return nil
+		}
+	}
+
+	t.Phase = InitialBackupCompleted
+
+	// Annotate persistent storage resources with PV actions
+	// This will also return the number of pods we have annotated to be backed up
+	// by restic. This is useful for knowing whether or not we need to create
+	// staging pods
+	err, resticPodCount := t.annotateStorageResources()
```

Comment:
Ah, finally found what `resticPodCount` means. Thoughts on `resticAnnotationCount` instead? Now that I know what this is I don't think it will confuse me, but initially thought this was referring to the number of actual `restic` pods existing on the cluster.